src/apps/Altinn.AccessManagement/src/Altinn.AccessMgmt.Core/Utils/IngestService.cs

```diff
@@ -0,0 +1,253 @@
+            throw new ArgumentNullException(nameof(T));
+        }
+
+        var tableName = GetTableName<T>(model);
+        var ingestColumns = GetColumns<T>(model);
+
+        return await WriteToIngest(data, ingestColumns, tableName.TableName, cancellationToken);
+    }
+
+    /// <inheritdoc />
+    public async Task<int> IngestTempData<T>(List<T> data, Guid ingestId, CancellationToken cancellationToken = default)
+    {
+        if (ingestId.Equals(Guid.Empty))
+        {
+            throw new Exception(string.Format("Ingest id '{0}' not valid", ingestId.ToString()));
+        }
+
+        var table = GetTableName<T>(dbContext.Model);
+        var ingestColumns = GetColumns<T>(dbContext.Model);
+
+        string columnStatement = string.Join(',', ingestColumns.Select(t => t.Name));
+
+        var ingestName = ingestId.ToString().Replace("-", string.Empty);
+        string ingestTableName = "ingest." + table.TableName + "_" + ingestName;
+
+        var createIngestTable = $"CREATE UNLOGGED TABLE IF NOT EXISTS {ingestTableName} AS SELECT {columnStatement} FROM {table.SchemaName}.{table.TableName} WITH NO DATA;";
+
+        await dbContext.Database.ExecuteSqlRawAsync(createIngestTable, cancellationToken);
+
+        var completed = await WriteToIngest(data, ingestColumns, ingestTableName, cancellationToken);
+
+        return completed;
+    }
+
+    /// <inheritdoc />
+    public async Task<int> MergeTempData<T>(Guid ingestId, AuditValues auditValues, IEnumerable<string> matchColumns = null, CancellationToken cancellationToken = default)
+    {
+        if (matchColumns == null || matchColumns.Count() == 0)
+        {
+            matchColumns = ["id"];
+        }
+
+        var table = GetTableName<T>(dbContext.Model);
+        var ingestColumns = GetColumns<T>(dbContext.Model);
+
+        string columnStatement = string.Join(',', ingestColumns.Select(t => t.Name));
+
+        var ingestName = ingestId.ToString().Replace("-", string.Empty);
+        string ingestTableName = "ingest." + table.TableName + "_" + ingestName;
+
+        var mergeMatchStatement = string.Join(" AND ", matchColumns.Select(t => $"(target.{t} IS NULL AND source.{t} IS NULL OR target.{t} = source.{t})"));
+        var mergeUpdateUnMatchStatement = string.Join(
+            " OR ",
+            ingestColumns
+                .Where(t => matchColumns.Count(y => y.Equals(t.Name, StringComparison.OrdinalIgnoreCase)) == 0)
+                .Select(t =>
+                    $"(" +
+                    $"target.{t.Name} <> source.{t.Name} " +
+                    $"OR (target.{t.Name} IS NULL AND source.{t.Name} IS NOT NULL) " +
+                    $"OR (target.{t.Name} IS NOT NULL AND source.{t.Name} IS NULL)" +
+                    $")"
+                )
+        );
+
+        string mergeUpdateStatement = string.Join(", ", ingestColumns.Where(t => !matchColumns.Any(y => y.Equals(t.Name, StringComparison.OrdinalIgnoreCase))).Select(t => $"{t.Name} = source.{t.Name}"));
```

Comment:
i eksisterende PostgresIngestService måtte vi inn med ett unntak her for RolePackage, for å ikke forsøke å oppdatere primarykey på updates, pga. foreignkey constraint vi har på disse fra DelegationPackage.

Strengt tatt vil jeg anta at det egentliig burde være sant for alle updates, at man beholder original primarykey.